### PR TITLE
haxe 4.1.0

### PR DESCRIPTION
**Pull request by vladimyr**
Update test with cpp compilation example.

### CODE REVIEW

File: Formula/haxe.rb [hunk 1]
```diff
@@ -1,9 +1,8 @@
 class Haxe < Formula
   desc "Multi-platform programming language"
   homepage "https://haxe.org/"
-  url "https://github.com/HaxeFoundation/haxe.git",
-      :tag      => "4.0.5",
-      :revision => "67feacebc59d5cd9dce232058f12cb933622d619"
+  url "https://github.com/HaxeFoundation/haxe/archive/4.1.0.tar.gz"
+  sha256 "c815b0671b0a679faab226464fe4f75c9868c5084a39073f6698ebc27e1455b1"
```
Comment (andyli) [inline Formula/haxe.rb:5]: We need to use `:tag` because there is git submodules in the Haxe repo. Git submodules are absent from the GitHub archive.

Comment (vladimyr) [reply Formula/haxe.rb:5]: TIL thanks!

Comment (vladimyr) [reply Formula/haxe.rb]: It built successfully on a local machine, let's see what CI thinks about it. 🤞

**Comment by vladimyr** [comment]
I'm not sure what is causing 10.14 to fail but quick web search reveals that it looks suspiciously similar to https://github.com/giampaolo/psutil/issues/1632 🤔

**Comment by andyli** [comment]
It could be a hxcpp issue. Anyway, the haxe compiler ran without problem.

I think it's better to remove the new cpp test since it involves hxcpp which is not in scope of this formula. Notice the hxcpp version is not pinned, which makes the test non-deterministic too.

**Comment by vladimyr** [comment]
> I think it's better to remove the new cpp test since it involves hxcpp which is not in scope of this formula. Notice the hxcpp version is not pinned, which makes the test non-deterministic too.

That is true but it also resembles what end-users will end up doing. Let's try with `js` target instead. It doesn't require external deps and still represents a valid example of real-world usage.

**Review by bayandin — APPROVED** [review]
Well done @vladimyr and @andyli, thanks!

**Pull request closed without merging**